src/Util/Profiler.php

```diff
@@ -0,0 +1,235 @@
+	 */
+	private $enabled;
+	/**
+	 * @var bool True, if the Profiler should measure the whole rendertime including functions
+	 */
+	private $rendertime;
+
+	/**
+	 * @var LoggerInterface The profiler logger
+	 */
+	private $logger;
+
+	/**
+	 * @param LoggerInterface $logger The profiler logger
+	 * @param bool $enabled           True, if the Profiler is enabled
+	 * @param bool $renderTime        True, if the Profiler should measure the whole rendertime including functions
+	 */
+	public function __construct(LoggerInterface $logger, $enabled = false, $renderTime = false)
+	{
+		$this->enabled = $enabled;
+		$this->rendertime = $renderTime;
+		$this->logger = $logger;
+		$this->reset();
+	}
+
+	/**
+	 * Saves a timestamp for a value - f.e. a call
+	 * Necessary for profiling Friendica
+	 *
+	 * @param int $timestamp the Timestamp
+	 * @param string $value A value to profile
+	 * @param string $callstack The callstack of the current profiling data
+	 */
+	public function saveTimestamp($timestamp, $value, $callstack = '')
+	{
+		if (!$this->enabled) {
+			return;
+		}
+
+		$duration = (float) (microtime(true) - $timestamp);
+
+		if (!isset($this->performance[$value])) {
+			// Prevent ugly E_NOTICE
+			$this->performance[$value] = 0;
+		}
+
+		$this->performance[$value] += (float) $duration;
+		$this->performance['marktime'] += (float) $duration;
+
+		if (!isset($this->callstack[$value][$callstack])) {
+			// Prevent ugly E_NOTICE
+			$this->callstack[$value][$callstack] = 0;
+		}
+
+		$this->callstack[$value][$callstack] += (float) $duration;
+	}
+
+	/**
+	 * Resets the performance and callstack profiling
+	 *
+	 * @param bool $performance If true, reset the performance (Default true)
+	 * @param bool $callstack   If true, reset the callstack (Default true)
+	 */
+	public function reset($performance = true, $callstack = true)
+	{
+		if ($performance) {
+			$this->performance = [];
+			$this->performance['start'] = microtime(true);
+			$this->performance['database'] = 0;
+			$this->performance['database_write'] = 0;
+			$this->performance['cache'] = 0;
+			$this->performance['cache_write'] = 0;
+			$this->performance['network'] = 0;
+			$this->performance['file'] = 0;
+			$this->performance['rendering'] = 0;
+			$this->performance['parser'] = 0;
+			$this->performance['marktime'] = 0;
+			$this->performance['marktime'] = microtime(true);
+		}
+
+		if ($callstack) {
+			$this->callstack['database'] = [];
+			$this->callstack['database_write'] = [];
+			$this->callstack['cache'] = [];
+			$this->callstack['cache_write'] = [];
+			$this->callstack['network'] = [];
+			$this->callstack['file'] = [];
+			$this->callstack['rendering'] = [];
+			$this->callstack['parser'] = [];
+		}
+	}
+
+	/**
+	 * Save the current profiling data to a log entry
+	 *
+	 * @param string $message Additional message for the log
+	 */
+	public function saveLog($message = '')
+	{
+		// Write down the performance values into the log
+		if ($this->enabled) {
```

Comment:
Please return early instead.
```suggestion
		if (!$this->enabled) {
			return;
		}
```